### PR TITLE
little oprimization on copy handler

### PR DESCRIPTION
**Pull request by reshke**
here we are like start a batch of goroutines, wait until all of them exec and start another batch
	 which is not most effitient way to do parralel work

### CODE REVIEW

File: internal/copy_handler.go [hunk 1]
```diff
@@ -33,50 +31,53 @@ func HandleCopy(fromConfigFile string, toConfigFile string, backupName string, w
...
-			return false, err
+	maxParallelJobsCount := 8
+
+	tickets := make(chan interface{}, maxParallelJobsCount)
+
+	for t := 0; t < maxParallelJobsCount; t++ {
+		tickets <- nil
+	}
+
+	errors := make(chan error)
```
Comment (defntvdm) [inline internal/copy_handler.go:42]: Make it buffered and read all errors after cycle

Comment (reshke) [reply internal/copy_handler.go]: sure bro :ok_hand: :sunglasses: :100:

File: internal/pagefile_new.go [hunk 2]
```diff
@@ -201,4 +201,4 @@ func GetIncrementHeaderFields(increment io.Reader) (uint64, uint32, []byte, erro
 		return 0, 0, nil, err
 	}
 	return fileSize, diffBlockCount, diffMap, nil
-}
\ No newline at end of file
+}
```
Comment (reshke) [inline internal/pagefile_new.go:204]: make apply fmt

File: internal/wal_segment_runner_test.go [hunk 3]
```diff
@@ -102,7 +102,7 @@ func TestWalSegmentRunner_SwitchTimelinesMissingSegments(t *testing.T) {
 		},
 	}
 
-	expectedFoundSegmentsByTimeline := map[uint32][]internal.WalSegmentNo {
+	expectedFoundSegmentsByTimeline := map[uint32][]internal.WalSegmentNo{
```
Comment (reshke) [inline internal/wal_segment_runner_test.go:105]: make fmt

**Review by x4m — APPROVED** [review]

File: internal/copy_handler.go [hunk 4]
```diff
@@ -33,50 +32,59 @@ func HandleCopy(fromConfigFile string, toConfigFile string, backupName string, w
...
+
+		go func(info CopyingInfo) {
+			defer wg.Done()
+			err := copyObject(info)
+			tickets <- nil
+			errors <- err
+		}(info)
 	}
+
+	wg.Wait()
```
Comment (defntvdm) [inline internal/copy_handler.go:67]: It looks like some errors are ignored here

Comment (reshke) [reply internal/copy_handler.go:73]: sure, fixed

**Review by defntvdm — APPROVED** [review]

**Pull request merged**